### PR TITLE
Refactor bastion methods around scope and remove region from status

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vince@vincepri.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/aws/actuators/machine_scope.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	}
+
+	machineStatus, err := v1alpha1.MachineStatusFromProviderStatus(params.Machine.Status.ProviderStatus)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get machine provider status")
+	}
+
+	var machineClient client.MachineInterface
+	if params.Client != nil {
+		machineClient = params.Client.Machines(params.Cluster.Namespace)
```
[the remark this answers is not in the document]
Comment (vincepri) [reply pkg/cloud/aws/actuators/machine_scope.go]: Either one could be used, I can switch it to the machine one if preferred. In any case, they are supposed to be the same.

[the remark this answers is not in the document]
Comment (detiber) [reply pkg/cloud/aws/actuators/machine_scope.go]: @vincepri lets update to use the machine namespace here, that way it'll be easier to decouple once there is a better upstream solution for linking clusters to machines.

[the remark this answers is not in the document]
Comment (vincepri) [reply pkg/cloud/aws/actuators/machine_scope.go]: Done

**Pull request merged**